ReactiveCocoa/Swift/Property.swift

```diff
@@ -89,35 +105,71 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	/// created from the `values` producer.
 	public var value: Value {
 		get {
-			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			return withValue { $0 }
 		}
 
 		set {
-			lock.lock()
-			_value = newValue
-			observer.sendNext(newValue)
-			lock.unlock()
+			modify { _ in newValue }
 		}
 	}
 
+	/// A signal that will send the property's changes over time,
+	/// then complete when the property has deinitialized.
+	public let signal: Signal<Value, NoError>
+
 	/// A producer for Signals that will send the property's current value,
 	/// followed by all changes over time, then complete when the property has
 	/// deinitialized.
-	public let producer: SignalProducer<Value, NoError>
+	public var producer: SignalProducer<Value, NoError> {
```

Comment:
It should be still the source of truth. When a producer is being started, it acquires the recursive lock and then send the current value and setup the forwarding within the same critical section. So it should be consistent.